From 2232b6a11fb2f319eb5abd1a7e8646c5a677c84f Mon Sep 17 00:00:00 2001
From: ole <gitlab+account@ole.mn>
Date: Sat, 10 Dec 2022 20:15:48 +0100
Subject: [PATCH] mono theme template

---
 src/kgx-terminal.c | 40 ++++++++++++++++++++--------------------
 1 file changed, 20 insertions(+), 20 deletions(-)

diff --git a/src/kgx-terminal.c b/src/kgx-terminal.c
index 26ca81d..11d859d 100644
--- a/src/kgx-terminal.c
+++ b/src/kgx-terminal.c
@@ -115,22 +115,22 @@ update_terminal_colors (KgxTerminal *self)
 
   // Workings of GDK_RGBA prevent this being static
   GdkRGBA palette[16] = {
-    GDK_RGBA ("241f31"), // Black
-    GDK_RGBA ("c01c28"), // Red
-    GDK_RGBA ("2ec27e"), // Green
-    GDK_RGBA ("f5c211"), // Yellow
-    GDK_RGBA ("1e78e4"), // Blue
-    GDK_RGBA ("9841bb"), // Magenta
-    GDK_RGBA ("0ab9dc"), // Cyan
-    GDK_RGBA ("c0bfbc"), // White
-    GDK_RGBA ("5e5c64"), // Bright Black
-    GDK_RGBA ("ed333b"), // Bright Red
-    GDK_RGBA ("57e389"), // Bright Green
-    GDK_RGBA ("f8e45c"), // Bright Yellow
-    GDK_RGBA ("51a1ff"), // Bright Blue
-    GDK_RGBA ("c061cb"), // Bright Magenta
-    GDK_RGBA ("4fd2fd"), // Bright Cyan
-    GDK_RGBA ("f6f5f4"), // Bright White
+    GDK_RGBA ("000000"), // Black
+    GDK_RGBA ("cc0000"), // Red
+    GDK_RGBA ("4e9a06"), // Green
+    GDK_RGBA ("c4a000"), // Yellow
+    GDK_RGBA ("3465a4"), // Blue
+    GDK_RGBA ("75507b"), // Magenta
+    GDK_RGBA ("06989a"), // Cyan
+    GDK_RGBA ("d3d7cf"), // White
+    GDK_RGBA ("555753"), // Bright Black
+    GDK_RGBA ("ef2929"), // Bright Red
+    GDK_RGBA ("8ae234"), // Bright Green
+    GDK_RGBA ("fce94f"), // Bright Yellow
+    GDK_RGBA ("729fcf"), // Bright Blue
+    GDK_RGBA ("ad7fa8"), // Bright Magenta
+    GDK_RGBA ("34e2e2"), // Bright Cyan
+    GDK_RGBA ("eeeeec"), // Bright White
   };
 
   if (self->theme == KGX_THEME_AUTO) {
@@ -151,14 +151,14 @@ update_terminal_colors (KgxTerminal *self)
       bg = (GdkRGBA) { 0.05, 0.05, 0.05, 0.96 };
       break;
     case KGX_THEME_DAY:
-      fg = (GdkRGBA) { 0.0, 0.0, 0.0, 0.0 };
-      bg = (GdkRGBA) { 1.0, 1.0, 1.0, 1.0 };
+      fg = (GdkRGBA) { 1.000, 1.000, 1.000, 1.0 };
+      bg = (GdkRGBA) { 0.000, 0.000, 0.000, 0.95 };
       break;
     case KGX_THEME_NIGHT:
     case KGX_THEME_AUTO:
     default:
-      fg = (GdkRGBA) { 1.0, 1.0, 1.0, 1.0};
-      bg = (GdkRGBA) { 0.05, 0.05, 0.05, 0.96 };
+      fg = (GdkRGBA) { 1.000, 1.000, 1.000, 1.0 };
+      bg = (GdkRGBA) { 0.000, 0.000, 0.000, 0.95 };
       break;
   }
 
-- 
2.38.1

